tst/testinstall/trans.tst

```diff
@@ -749,25 +749,26 @@ false
 gap> IsInjectiveListTrans([65536], ImageListOfTransformation(f));
 true
 gap> IsInjectiveListTrans(1, f);
-Error, the first argument must be a list (not a integer)
+Error, IsInjectiveListTrans: <list> must be a small list (not the integer 1)
 gap> IsInjectiveListTrans([1], 1);
-Error, the second argument must be a transformation or a list (not a integer)
+Error, IsInjectiveListTrans: <obj> must be a transformation or a list (not the\
+ integer 1)
 gap> IsInjectiveListTrans([1, 2], "def");
-Error, the second argument must consist of positive integers (not a character)
+Error, <obj> must be a list of positive small integers (not a character)
 gap> IsInjectiveListTrans(ID_TRANS4, "def");
-Error, the first argument must be a list (not a transformation (large))
+Error, IsInjectiveListTrans: <list> must be a small list (not a transformation\
+ (large))
 gap> IsInjectiveListTrans([1, 2], [2,3]);
-Error, the second argument must consist of positive integers in the range [1 .\
-. 2]
+Error, <obj> must be a list of positive small integers in the range [1 .. 2]
 gap> IsInjectiveListTrans([1, []], f);
-Error, the entries of the first argument must be positive integers (not a empt\
-y plain list)
+Error, <list> must be a list of positive small integers (not a empty plain lis\
```

Comment:
I was under the impression that the "not a empty plain list" article inflection was resolved by #3691. I guess this must be triggering some different code than used by #3691, right?

Reply:
Indeed. There are still some places in the kernel which don't use `RequireArgumentEx` to print error messages involving `(not a %s)`, most of them in `trans.cc`. This PR actually replaces some more of these(`git grep -F '(not a ' src | wc -l` give me 40 on `master` and 27 with this PR). and then there are a few where there is a good reason for that. E.g. in some of them, the error message contains an other `%d` print formatter, and that currently can't be handed by `RequireArgumentEx`. Example from `trans.cc`:
```C
        if (!IS_INTOBJ(ELM_LIST(ran, i))) {
            ErrorQuit("TransformationListListNC: <ran>[%d] must be a small "
                      "integer (not a %s)",
                      (Int)i, (Int)TNAM_OBJ(ELM_LIST(ran, i)));
        }
```
I guess we could add another helper to deal with this particular use case, though I think it's really only a very small number, so may not be worth it.

Reply:
I was just curious, thanks for the explanation 🙂 As far as I'm concerned I agree, it's probably not worth it.